src/amo/withExperiment.js

```diff
@@ -175,15 +181,79 @@ export const withExperiment =
         WrappedComponent,
       )})`;
 
-      componentDidMount() {
-        const { _getVariant, _isExperimentEnabled, cookies, id, variants } =
-          this.props;
-
+      initialExperimentSetup(props) {
+        const {
+          _getVariant,
+          _isExperimentEnabled,
+          dispatch,
+          id,
+          storedVariant,
+          variants,
+        } = props;
+
+        let { variant } = this;
         const isEnabled = _isExperimentEnabled({ _config, id });
         const registeredExperiments = this.getExperiments();
         const experimentInCookie = this.cookieIncludesExperiment(
           registeredExperiments,
         );
+        const addExperimentToCookie = !experimentInCookie;
+        const variantFromStore = storedVariant;
```

Comment:
Maybe you could inject `variantFromStore` instead of `storedVariant`.

Reply:
(that being said, `storedVariant` _stored_ in a redux _store_ sounds a bit redundant to me. I believe the redux store could hold an experiment state made of experiment IDs and variants)

Reply:
I have changed this a bit, because I am now storing multiple variants, one per experiment, in the state. If the newer code (which I will push in a bit) doesn't address this comment please let me know.